### DISCUSSION TITLE
[BUG] Unable to autocomplete Option or Argument with Path type

### DISCUSSION THREAD

**Discussion by l-vincent-l** [fastapi/typer#663] [Questions]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the Typer documentation, with the integrated search.
- [X] I already searched in Google "How to X in Typer" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to Typer but to [Click](https://github.com/pallets/click).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
# docs_src/multiple_values/arguments_with_multiple_values/tutorial001.py
from pathlib import Path
from typing import List

import typer


def main(files: List[Path], celebration: str):
    for path in files:
        if path.is_file():
            print(f"This file exists: {path.name}")
            print(celebration)


if __name__ == "__main__":
    typer.run(main)
```

### Description

When I try to run the tutorial I expect to have autocompletion on `Path` arguments, but when I hit tab a space is added and there is no autocompletion.

The workaround I found is to set the type to `str` instead of `Path` but I'm not satisfied with it.

[Capture vidéo du 12-09-2023 18:06:39.webm](https://github.com/tiangolo/typer/assets/2757318/66fddc78-bece-4dca-abcd-2cc019459eb1)

### Operating System

Linux

### Operating System Details

I use bash

### Typer Version

0.9.0

### Python Version

3.10.12

**Comment by YuriiMotov** [comment]
Can't reproduce on (tried on Ubuntu, bash)..

Typing `typer tutorial001.py run tu[TAB]` results in
```
(.venv) yurii@yurii-laptop:~/code/typer/motov/docs_src/multiple_values/arguments_with_multiple_values$ typer tutorial001.py run tutorial00
```
one more TAB
```
(.venv) yurii@yurii-laptop:~/code/typer/motov/docs_src/multiple_values/arguments_with_multiple_values$ typer tutorial001.py run tutorial00
tutorial001.py     tutorial002_an.py  tutorial002.py
```


I'll close this as it was probably fixed somewhere.
If you still experience this, please comment here or open new discussion
